packages/cli-kit/src/constants.ts

```diff
@@ -28,6 +28,11 @@ const constants = {
     verbose: 'SHOPIFY_FLAG_VERBOSE',
     noAnalytics: 'SHOPIFY_CLI_NO_ANALYTICS',
   },
+  logStreams: {
+    cli: 'shopify.cli.log',
+    createApp: 'shopify.create-app.log',
+    createHydrogen: 'shopify.create-hydrogen.log',
+  },
```

Comment:
What would be the value as a user to have multiple logfiles? What about logging to a single file and add a prefix that allows grep-like filtering: `shopify log --filter hydrogen`

Reply:
The logs are pretty noisy as is. I think having multiple files is a benefit, but maybe this is a larger DX discussion.

Note that cli for apps and hydrogen are still in 1 log, which maybe we'd want to change, hard to say but also would be difficult to implement as it's done per CLI run function and we don't have differentiation in CLI.

Nothing has changed in this PR, outside of adding the ability to choose a file in the command. If this is going to hold up the remainder of the work, then I'd say let's just skip it for now and follow up with @MeredithCastile

Reply:
I agree that having multiple streams could be confusing, and `create-app`/`create-hydrogen` would normally be called only once, no?

Reply:
Consulting @MeredithCastile @gfscott, do we see value in different logfiles for `create-app` and `create-hydrogen`? Or just unite into 1 logfile?

Do we want to just unite everything into 1 file now and brainstorm ways we might distinguish log lines in other ways? (For example, hydrogen commands vs app commands... though there are some technical hurdles to solve there)

Reply:
Sorry I missed this, @gfscott!

* I think it's something of an either-or, either they're written to 1 file or multiple files. But of course we can identify log lines per CLI/sub-CLI/command and differentiate based on that. It's all a question of what we care to do and how much we care vs. shipping actual CLI features.
* Currently it would be logged to a single file, not 1 file per run. That would create a massive number of files, and a big question of how/when to clean them up. The system for log rotation is extremely simple at the moment, and I'd like to keep it that way.